### PR TITLE
Fixes #39

### LINKED ISSUE #39: Avoid DRY in predict-eth helper functions
### Motivation

main2.md and most end-to-end READMEs each have their own copy of the helper functions. [Link from main2](https://github.com/oceanprotocol/predict-eth/blob/main/challenges/main2.md#appendix-load-helper-functions). (hack1.md and main1.md too, but they are obsolete so matter less)

What the functions do: create an ocean instance, convert amongst time units, compute target times, filter cex data to target times, load/save lists, and plot predicted vs actual. They've been very helpful; before them, it was code scattered throughout the README that bloated it.

But now that we have more examples, the DRY violations are hurting maintainability, debuggability, etc. For example, if we discover a bug in one README (eg nmse calc), we have to update several. Or if we improve one README (eg brownie support), we should improve several.

### TODOs

- [x] Identify candidate solutions (below)
- [x] Discuss & decide on a first solution. 
- [x] Implement first solution "cand 2"
- [ ] Implement second solution "cand 5" pypi of predict-eth

### Candidate solutions

**Cand 1**

What: Do nothing
- Pros: simple for us - we don't have to do anything. Simple-ish for the user. User understands the functions.
- Cons: DRY violations, which hurt maintainability & debuggability. User has to copy & paste a bunch of code.
- (We keep this as a candidate in case the "cons" of other solutions are even greater than this one.)

**Cand 2**

What: have a special readme called helpers.md, with code that users simply copy & paste the code. Other READMEs point to it (rather than having their own copy)
- Pros: simple for us: could implement in 30 min. Simple-ish for the user. User understands the functions.
- Cons: user has to go to another README. User has to copy & paste a bunch of code.

**Cand 3**

What: put the functionality into ocean.py
- Pros: simple for user. Simple for us.
- Cons: this is functionality that ocean.py really shouldn't have. And, any changes we make and want users to see, we'd have to do a new ocean.py release

**Cand 4**

What: get the user to `git clone` this repo. Then import the helper functions.
- Pros: simple for user. Simple for us.
- Cons: slightly messy due to asking user to git clone

**Cand 5**

What: create "predict-eth" pypi library from the helper functions. Have it via this repo, to keep everything together. Then at the beginning of a README, pip install this library too. 
- How: mirror [pybundlr Development section](https://github.com/oceanprotocol/pybundlr#development), which links to a release process and how developers develop. The release process is streamlined from pypi documentation. 
- Pros: simple for user. Relatively simple for us, given that we can mirror pybundlr
- Cons: takes _some_ effort to set up with pypi (1 day, by the time all is done)

**Cand 6**

What: create a "web3data" pypi library that includes the helper functions, ocean.py dependency, pybundlr dependency, and maybe more in future.
- How: like cand 5
- Pros & cons: like cand 5
- Additional pro: readily extensible beyond predict-eth

### Analysis / Recommendation

Trent Nov 14: 
- Never cand 3, doesn't make sense. Cand 2 > cand 1
- Do Cand 2 ASAP. Then Cand 4 next. Then maybe Cand 5 or Cand 6 or Cand 6 at some point.

### PR DESCRIPTION
**Pull request by jfdelgad**
Fixes #39  .

Changes proposed in this PR:

-  Move appendix code to a. separated md file
-  Remove apendix from main1 and main2
-

### CODE REVIEW

File: challenges/main1.md [hunk 1]
```diff
@@ -225,144 +225,4 @@ plot_prices(cex_vals, pred_vals)
...
-    ax.plot(x, pred_vals, '-', label="Pred. values")
-    ax.legend(loc='lower right')
-    plt.ylabel("ETH price")
-    plt.xlabel("Hour")
-    fig.set_size_inches(18, 18)
-    plt.xticks(x)
-    plt.show()
-
-```
+See [helpers.md](../examples/helpers.md) to load the helper functions.
```
Comment (trentmc) [inline challenges/main1.md:228]: Don't put this in the appendix. Rather, fully remove the appendix and put it in section 1.4 "load helper functions".

File: challenges/main2.md [hunk 2]
```diff
@@ -226,158 +226,6 @@ plot_prices(cex_vals, pred_vals)
 
 ## Appendix: Load helper functions
 
-If the Python console isn't already open: `python`
+See [helpers.md](../examples/helpers.md) to load the helper functions.
```
Comment (trentmc) [inline challenges/main2.md:229]: Don't put this in the appendix. Rather, fully remove the appendix and put it in section 1.4 "load helper functions".

**Review by trentmc — CHANGES_REQUESTED** [review]

File: challenges/main1.md (hunk 1 above)
Comment (jfdelgad) [reply challenges/main1.md:228]: Done

File: challenges/main1.md [hunk 3]
```diff
@@ -225,144 +225,4 @@ plot_prices(cex_vals, pred_vals)
 
 ## Appendix: Load helper functions
```
[the remark this answers is not in the document]
Comment (jfdelgad) [reply challenges/main1.md:227]: Done

**Pull request merged**